_case_studies/essayjack.markdown

```diff
@@ -13,21 +13,35 @@ project-url: https://www.essayjack.com
 status: published
 ---
 
-###### EssayJack is an essay templating platform. Created by professors, it provides a way for instructors to guide students through the daunting task of essay writing.
-
-## The Problem
+###### EssayJack is an essay templating platform that has an interactive essay editor that pre-structures student essays. Created by professors, it provides a way for instructors to guide students through the daunting task of essay writing.
 
+## The Challenge
+EssayJack founder Lindy came to us with a “traditional” Rails app that needed some intensive care. The app had been developed as an MVP, had gained some traction and had a user base. But key features of this beta version of the app behaved unpredictably, at times breaking the UX in a jarring way. The app’s test suite was not in a state where we could rely on it to provide the security we needed to conduct a refactoring exercise. Being a team who isn’t afraid of anything, we welcomed this opportunity. A Project Rescue operation began.
 
 ## Our Approach
+Our contribution to the project began with a thorough audit. This helped us familiarise ourselves with the codebase, and over the course of a month, we presented our findings to EssayJack. During this time, we also helped the existing development team fix critical bugs that were impeding the users’ experience on the app. This first month of work demonstrated our ability to manage such projects, giving EssayJack the assurance they needed to continue their engagement with us.
+
+The next order of business was stabilize the product. This stabilization process included a period of further bug fixing. We got the test suite into a state where it could be relied on. We expanded it’s coverage and hooked it into our CI process. To solve certain performance issues and make deployment more convenient, we migrated the app to AWS OpsWorks.
+
+With a reliable test suite in place, we began refactoring. The first major area we tackled was the template editor. Rebuilding this key feature was a crucial step in moving the product into a phase where it was open for extension. The refactoring put us in a good place for a front-end revamp to be considered.
+
+A design firm called Digital Natives were brought in at this point to design a brand new app for EssayJack. Our goal was to bring this reimagined EssayJack to life. The complex essay editor had a lot of moving parts, so we needed a solution that would give us a way to manage state and provide us with a view layer. We went with React.
```

Comment:
Link to Digital Natives homepage? Might help their SEO _ducks_

Reply:
Updated link